cupy/linalg/norms.py

```diff
@@ -57,7 +54,7 @@ def norm(x, ord=None, axis=None, keepdims=False):
             return abs(x).min(axis=axis, keepdims=keepdims)
         elif ord == 0:
             # Zero norm
-            return (x != 0).sum(axis=axis, keepdims=keepdims, dtype=x.dtype)
+            return (x != 0).sum(axis=axis, keepdims=keepdims, dtype='d')
```

Comment:
I think it is better to add a comment that why we do not convert to the dtype of `x` (Is it because NumPy does so?)